src/yz_index_hashtree.erl

```diff
@@ -639,7 +640,15 @@ get_all_locks(Type, Pid) ->
     %% file page cache but the bg manager stuff is kind of convoluted
     %% and there isn't time to figure this all out for 2.0. Thus,
     %% Yokozuna will not bother with the Solr lock for now.
-    ok == yz_entropy_mgr:get_lock(Type, Pid).
+    try yz_entropy_mgr:get_lock(Type, Pid) of
+        ok -> true;
+        Other ->
+            lager:debug("Could not get lock: ~p", [Other]),
+            false
+    catch exit:{timeout,_} ->
```

Comment:
Would it be so bad if we just had the yz_index_hashtree process die when it cannot get the lock?
The hashtree would be persisted to disk on termination.
That would allow the entropy_mgr to do its clean-up of held locks.